src/builder/QuarkBuilder.sol

```diff
@@ -278,6 +279,63 @@ contract QuarkBuilder {
             }
         }
 
+        // Check if need to wrap/unwrap token to cover the transferIntent amount
+        // If the existing balance is not enough to cover the transferIntent amount, wrap/unwrap the counterpart token here
+        uint256 existingBalance =
+            Accounts.getBalanceOnChain(transferIntent.assetSymbol, transferIntent.chainId, chainAccountsList);
+        if (
+            existingBalance < transferIntent.amount
+                && TokenWrapper.hasWrapperContract(transferIntent.chainId, transferIntent.assetSymbol)
+        ) {
```

Comment:
I guess this logic will also have to be applied to supply right? Let's leave a TODO if we plan on doing that separately

Reply:
Sure will add `TODO:` 👍 
iirc we also have added `TODO` in the supply PR there as well.  It's always good to have more notes than less.